### PR TITLE
[7.17](backport #34376) [Heartbeat] Fix hint-based monitor gen

### PR DESCRIPTION
**Pull request by gsantoro**
## What does this PR do?

Backport [7.17](backport #34376) [Heartbeat] Fix hint-based monitor gen

## Why is it important?

Backport a bug that didn't allow to correctly create heartbeat configs

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by gsantoro** [comment]
hello @emilioalvap,
do you mind having a good look at this backport PR for 7.17. I tried to do my best but I am not very familiar with the code.

Do you mind any of you two (or both) @MichaelKatsoulis or @ChrsMark to have a good look at this PR.

**Review by andrewvc — APPROVED** [review]
LGTM, it's irritating that the linter is asking to fix code that wasn't modified here, but I guess those lines need tweaks.

**Review by ChrsMark — APPROVED** [review]
Looking at the diffs with the original PR, this one looks a valid backrport. (only diff is an empty line at https://github.com/elastic/beats/pull/34376/files#diff-b938fbca1dd039fb951f822ffb0b1e9f8e785eae12edf13853725f84b1dc46bcR71).

I'm not super familiar with the monitors' hints but I guess since the original PR was landed by the respective team this one sth we want as well.

LGTM!

**Review by gizas — APPROVED** [review]

**Review by MichaelKatsoulis — APPROVED** [review]

File: heartbeat/autodiscover/builder/hints/monitors.go [hunk 1]
```diff
@@ -32,14 +32,15 @@ import (
 )
 
 func init() {
-	autodiscover.Registry.AddBuilder("hints", NewHeartbeatHints)
+	_ = autodiscover.Registry.AddBuilder("hints", NewHeartbeatHints)
 }
 
 const (
 	montype    = "type"
```
Comment (emilioalvap) [inline heartbeat/autodiscover/builder/hints/monitors.go:39]:
```suggestion
```

**Pull request merged**